File: 3-Sum-Closest.js
/**
 * @param {number[]} arr
 * @param {number} target
 * @returns {number}
 */

class Solution {
    closest3Sum(nums, target) {
        // code here
        nums.sort(((a, b) => a - b))
        let n=nums.length
        let closestS= nums[0]+nums[1]+nums[2]
        for (let i=0; i<n-1;i++){
            if (i>0 & nums[i]===nums[i-1]){
                continue
            }
            let low=i+1
            let high=n-1
            
            while (low < high){
                let curS=nums[i]+nums[low]+nums[high]
                if (curS===target){
                    return curS
                }
                if (Math.abs(target- curS ) < Math.abs(target-closestS )||(Math.abs(curS - target) == Math.abs(closestS - target) & curS > closestS) ){
                    closestS = curS
                }
                if (curS<target){
                    low++
                }
                else{
                    high--
                }
            }
        }
        return closestS
        
    }
}

File: 3 Sum Closest.py
class Solution:
    def closest3Sum(self, arr, target):
        # code here
        arr.sort()
        n=len(arr)
        closestS=arr[0]+arr[1]+arr[2]
        for i in range(n-2):
            if i>0 and arr[i]==arr[i-1]:
                continue
            low,high=i+1,n-1
            while low<high:
                curS=arr[i]+arr[low]+arr[high]
                if curS==target:
                    return curS
                if abs(curS - target) < abs(closestS - target) or (abs(curS - target) == abs(closestS - target) and curS > closestS):
                    closestS = curS
                
                if curS<target:
                    low+=1
                else:
                    high-=1
        return closestS
